pkgs/by-name/wa/waybar-lyric/package.nix

```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule rec {
+  pname = "waybar-lyric";
+  version = "0.10.0";
+
+  src = fetchFromGitHub {
+    owner = "Nadim147c";
+    repo = "waybar-lyric";
+    rev = "v${version}";
+    sha256 = "sha256-glY0W/fB57pYrAhIxYSnkJBmqeN6vy0cwrS71CnKeK4=";
+  };
+
+  # Vendor hash for Go dependencies
+  vendorHash = "sha256-DBtSC+ePl6dvHqB10FyeojnYoT3mmsWAnbs/lZLibl8=";
+
+  # Disable tests
```

Comment:
Why?

Reply:
If it's failing to create a cache directory, that probably means it is trying to write to `~/.cache`

Usually we can avoid this by setting `export XDG_CACHE_HOME=$(mktemp -d)` in `preCheck`. If it doesn't support xdg, you can also make all of HOME writable using `writableTmpDirAsHomeHook`.

This is mentioned in a few places throughout the nixpkgs manual, but not in much detail. E.g. "common issues" under https://nixos.org/manual/nixpkgs/stable/#automatic-tests